pkg/manager/status/status_manager.go

```diff
@@ -0,0 +1,231 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	flowslatest "github.com/netobserv/network-observability-operator/api/v1beta2"
+	"github.com/netobserv/network-observability-operator/controllers/constants"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ComponentName string
+
+const (
+	FlowCollectorLegacy ComponentName = "FlowCollectorLegacy"
+	Monitoring          ComponentName = "Monitoring"
+)
+
+var allNames = []ComponentName{FlowCollectorLegacy, Monitoring}
+
+type Manager struct {
+	sync.Mutex
+
+	statuses map[ComponentName]ComponentStatus
```

Comment:
can u use syncMap here ?

Reply:
sounds good ... sometimes I don't like using sync.Map as the API is less user-friendly, and they don't always offer better performances ; but I just tried here and it plays nicely